src/main/java/amidst/mojangapi/minecraftinterface/legacy/LegacyMinecraftInterface.java

```diff
@@ -63,33 +62,26 @@ public LegacyMinecraftInterface(Map<String, SymbolicClass> symbolicClassMap, Rec
 			recognisedVersion);
 	}
 
+	@Override
+	public synchronized MinecraftInterface.WorldConfig createWorldConfig() throws MinecraftInterfaceException {
+		try {
+			initializeIfNeeded();
+			return new WorldConfig();
+		} catch (IllegalArgumentException e) {
+			throw new MinecraftInterfaceException("unable to create config", e);
+		}
+	}
+
 	// Only one thread can manipulate the Minecraft int cache at a time
 	private synchronized int[] getBiomeData(int x, int y, int width, int height, Object biomeGenerator)
 			throws MinecraftInterfaceException {
-		if (isIntCacheInUse) {
-			throw new IllegalStateException("This thread is already generating biome data");
-		}
-		isIntCacheInUse = true;
 		try {
 			resetIntCacheMethod.invokeExact();
-			return (int[]) getIntsMethod.invokeExact(biomeGenerator, x, y, width, height);
+			int[] biomeInts = (int[]) getIntsMethod.invokeExact(biomeGenerator, x, y, width, height);
+			// we have to clone the array so we aren't using a reference to an array that's still in the IntCache
+			return biomeInts.clone();
```

Comment:
Instead of cloning the `biomeInts` array, it would be more efficient to take the `biomeDataMapper` as a parameter and call it's `apply` method directly. This should be thread-safe, given that we are in a `synchronized` block and nobody else can access the `biomeInts` array.

Reply:
Fixed this by doing what you said.